File: stvs/src/main/java/edu/kit/iti/formal/stvs/model/config/ColumnConfig.java
package edu.kit.iti.formal.stvs.model.config;

import javafx.beans.property.DoubleProperty;
import javafx.beans.property.SimpleDoubleProperty;

/**
 * Configuration for table column. Contains GUI-related information about a column.
 *
 * @author Philipp
 */
public class ColumnConfig {
  private DoubleProperty width;

  /**
   * Default ColumnConfig.
   */
  public ColumnConfig() {
    width = new SimpleDoubleProperty(100);
  }

  /**
   * Create a new ColumnConfig.
   *
   * @param colwidth The initial column width
   */
  public ColumnConfig(double colwidth) {
    if (colwidth <= 0) {
      throw new IllegalArgumentException("Column width must be positive");
    }
    width = new SimpleDoubleProperty(colwidth);
  }

  /**
   * Get the current column width.
   *
   * @return The current column width
   */
  public double getWidth() {
    return width.get();
  }

  /**
   * Set the current column width.
   *
   * @param width The current column width
   */
  public void setWidth(double width) {
    if (width <= 0) {
      throw new IllegalArgumentException("Column width must be positive");
    }
    this.width.set(width);
  }

  @Override
  public boolean equals(Object obj) {
    if (this == obj) {
      return true;
    }
    if (!(obj instanceof ColumnConfig)) {
      return false;
    }

    ColumnConfig that = (ColumnConfig) obj;

    return width.get() == that.width.get();
  }

  @Override
  public int hashCode() {
    long bits = Double.doubleToRawLongBits(width.get());
    return (int) (bits ^ (bits >> 32));
  }

  public DoubleProperty widthProperty() {
    return width;
  }
}

File: stvs/src/main/kotlin/edu/kit/iti/formal/stvs/model/config/ColumnConfig.kt
/* *****************************************************************
 * This file belongs to verifaps-lib (https://verifaps.github.io).
 * SPDX-License-Header: GPL-3.0-or-later
 * 
 * This program isType free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program isType distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a clone of the GNU General Public
 * License along with this program.  If not, see
 * <http://www.gnu.org/licenses/gpl-3.0.html>.
 * *****************************************************************/
package edu.kit.iti.formal.stvs.model.config

import javafx.beans.property.SimpleDoubleProperty

/**
 * Configuration for table column. Contains GUI-related information about a column.
 *
 * @author Philipp
 */
class ColumnConfig {
    private val widthProperty = SimpleDoubleProperty(100.0)
    var width: Double
        get() = widthProperty.get()
        set(value) {
            require(value >= 0)
            widthProperty.set(value)
        }

    /**
     * Create a new ColumnConfig.
     *
     * @param colwidth The initial column width
     */
    constructor(colwidth: Double = 100.0) {
        require(!(colwidth <= 0)) { "Column width must be positive" }
        width = colwidth
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is ColumnConfig) return false

        if (width != other.width) return false

        return true
    }

    override fun hashCode(): Int = width.hashCode()
}